assets/js-image-loading.js

```diff
@@ -19,124 +21,329 @@ const handleImageLoading = () => {
     }
   }
 
-  const wrappers = document.querySelectorAll(`.${imageOptions.classes.wrapper}`);
+  const wrappers = document.querySelectorAll(`.${config.classes.wrapper}`),
+        mobileWidth = 992; // Define mobile width for responsive design
+
   if (!wrappers.length) return false;
 
-  const isInViewport = (el) => {
-    const rect = el.getBoundingClientRect();
-    return (
-      rect.top >= 0 &&
-      rect.left >= 0 &&
-      rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
-      rect.right <= (window.innerWidth || document.documentElement.clientWidth)
-    )
+  let observer = null;
+
+  // Helper function to decode lazy-loaded images
+  // Only used for images that are lazy-loaded but become visible
+  const decodeImage = (image) => {
+    if (!('decode' in image)) return;
+
+    image.decode().catch(() => {
+      // Silently fail - the image will still display normally
+    })
   }
 
   const sourcesDataLoad = (mainImage) => {
-    const wrapper = mainImage.closest(`.${imageOptions.classes.wrapper}`),
-          sources = wrapper.querySelectorAll(`source[${imageOptions.attributes.sourceSrcset}]`);
+    const wrapper = mainImage.closest(`.${config.classes.wrapper}`),
+          sources = wrapper.querySelectorAll(`source[${config.attributes.sourceSrcset}]`),
+          inViewport = $.is($.inViewport, 'function') ? $.inViewport(mainImage) : false,
+          slowConnection = $.slowConnection();
```

Comment:
isn't better for readability to declare each variable in a new line?

```suggestion
    const wrapper = mainImage.closest(`.${config.classes.wrapper}`);
    const sources = wrapper.querySelectorAll(`source[${config.attributes.sourceSrcset}]`),
    const inViewport = $.is($.inViewport, 'function') ? $.inViewport(mainImage) : false,
    const slowConnection = $.slowConnection();
```